pkg/server/server.go

```diff
@@ -183,15 +183,21 @@ func (s *ReplicationServer) WaitForShutdown() {
 }
 
 func (s *ReplicationServer) Shutdown() {
+	// cancel the context first so that all listeners/servers start shutting down
+	s.cancel()
```

Comment:
We'd want to make sure this doesn't break graceful shutdown of the API, since a lot of the dependencies of the API service will start to shut down once the context is canceled (for example, the metrics server, the node registry and registrant, and the sync service). I believe the DB will still be up because it is created with a parent context of the one being canceled.

Reply:
ok. its last again. I pulled out a few other services that needed explicit shutting down.